shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/repository/TokenRepositoryImpl.kt

```diff
@@ -0,0 +1,35 @@
+package co.nimblehq.blisskmmic.data.repository
+
+import co.nimblehq.blisskmmic.data.database.datasource.LocalDataSource
+import co.nimblehq.blisskmmic.data.database.model.TokenDatabaseModel
+import co.nimblehq.blisskmmic.data.network.datasource.NetworkDataSource
+import co.nimblehq.blisskmmic.data.network.target.LoginTargetType
+import co.nimblehq.blisskmmic.domain.model.Token
+import co.nimblehq.blisskmmic.domain.repository.TokenRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class TokenRepositoryImpl(
```

Comment:
I think `AuthenticationRepository` should be a better naming because we also have the login function here beside getting/saving the token, what do you think?